File: python-algorithm/leetcode/problem_1716.py
"""1716. Calculate Money in Leetcode Bank
https://leetcode.com/problems/calculate-money-in-leetcode-bank/
"""


def total_money(n: int) -> int:
    a, b = divmod(n, 7)
    start, end = 28, 28 + (a - 1) * 7
    ans = (start + end) * a // 2
    c = a + 1
    d = a + b
    return ans + (c + d) * b // 2

File: java-algo/src/main/java/com/leetcode/Problem1716.java
package com.leetcode;

public class Problem1716 {
    public int totalMoney(int n) {
        int x = n / 7, y = n % 7;
        return (7 * x + 49) * x / 2 + (1 + x + x + y) * y / 2;
    }
}
